### ISSUE TITLE
Exhale QPs with predicates results in an "empty.reduceLeft" error

### ISSUE THREAD

**Issue by tdardinier** [viperproject/carbon#391]
The Viper program

```
predicate p()

method m()
  ensures forall i: Int :: p()
```

results in the following error:

```
ERROR: The following exception occured in ViperServer: java.lang.UnsupportedOperationException: empty.reduceLeft
 trace:
  scala.collection.TraversableOnce.reduceLeft(TraversableOnce.scala:179)
  scala.collection.TraversableOnce.reduceLeft$(TraversableOnce.scala:177)
  scala.collection.mutable.ArrayBuffer.scala$collection$IndexedSeqOptimized$$super$reduceLeft(ArrayBuffer.scala:47)
  scala.collection.IndexedSeqOptimized.reduceLeft(IndexedSeqOptimized.scala:73)
  scala.collection.IndexedSeqOptimized.reduceLeft$(IndexedSeqOptimized.scala:72)
  scala.collection.mutable.ArrayBuffer.reduceLeft(ArrayBuffer.scala:47)
  scala.collection.TraversableOnce.reduce(TraversableOnce.scala:207)
  scala.collection.TraversableOnce.reduce$(TraversableOnce.scala:207)
  scala.collection.AbstractTraversable.reduce(Traversable.scala:104)
  viper.carbon.modules.impls.QuantifiedPermModule.translateExhale(QuantifiedPermModule.scala:723)
```

which occurs because a reduce operation is done on the arguments of the predicates, and in  this case the predicate has no argument.

This should probably be handled by the consistency checks.

**Comment by tdardinier** [comment]
[Corresponding Silicon issue](https://github.com/viperproject/silicon/issues/595)

**Comment by mschwerhoff** [comment]
Resolved in Silicon with PR https://github.com/viperproject/silicon/pull/597.

Corresponding regression in PR https://github.com/viperproject/silver/pull/554. The latter should only be accepted once this Carbon is was resolved.

**Comment by alexanderjsummers** [comment]
I can't reproduce the error in Carbon (I get a well-definedness error, but I think that's correct with the latest design)

**Comment by marcoeilers** [comment]
This no longer got us the error shown above, but resulted in Boogie code with quantifiers with no quantified variables, which led to a Boogie error. Fixed in PR #541.
